.github/workflows/enable-release-protection.yml

```diff
@@ -0,0 +1,55 @@
+#
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  SPDX-License-Identifier: Apache-2.0
+#
+
+# This workflow enables branch protection when a branch name containing 'release/' is created
+name: "enable release protection"
+on:
+  create
+
+jobs:
+  enable-release-branch-protection:
```

Comment:
@htztomic we will enable the branch protection for `release/*` branch by default and disable and enable it back after release branch has been created